rust/lance/src/datatypes/schema.rs

```diff
@@ -226,7 +226,23 @@ impl Schema {
         self.fields_pre_order().map(|f| f.id).collect()
     }
 
-    pub(crate) fn mut_field_by_id(&mut self, id: i32) -> Option<&mut Field> {
+    /// Get field by its id.
+    #[allow(dead_code)]
+    pub(crate) fn field_by_id(&self, id: impl Into<i32>) -> Option<&Field> {
+        let id = id.into();
```

Comment:
nit: should we cache the ids on the fly? other wise we have to redo the DFS over and over when we need multiple fields.

Reply:
It is not necessary, Multi-id  lookup can use `Schema::project_by_id([u32])`